### PR TITLE
raftexample: fix blocking on restart.

### PR DESCRIPTION
**Pull request by upccup**
Now when restart raftexample it will be blocked by the [readcommits](https://github.com/coreos/etcd/blob/master/contrib/raftexample/kvstore.go#L43) then the serveHttpKVAPI can not start

### CODE REVIEW

File: contrib/raftexample/kvstore_test.go [hunk 1]
```diff
@@ -1,47 +0,0 @@
-// Copyright 2016 The etcd Authors
```
Comment (gyuho) [inline contrib/raftexample/kvstore_test.go]: why delete this test?

Comment (upccup) [inline contrib/raftexample/kvstore_test.go]: now recoverFromSnapshot need snapshotter but is bother to mock

**Comment by gyuho** [comment]
@upccup Could you provide a way to reproduce that behavior? Restart works fine on my side?

Thanks!

**Comment by upccup** [comment]
@gyuho  first we should chang the [defaultSnapCount](https://github.com/coreos/etcd/blob/master/contrib/raftexample/raft.go#L73) smaller(mine is 10) to  trigger the dosnapshot. 

second use **curl -X PUT 127.0.0.1:12380 -D "1111"** 10 times. make sure the saveSnapshot was triggered.

then restart the raftexample

**Comment by gyuho** [comment]
@upccup I will take a closer look next week. Thanks!

File: contrib/raftexample/raft.go [hunk 2]
```diff
@@ -239,9 +239,8 @@ func (rc *raftNode) replayWAL() *wal.WAL {
 	// send nil once lastIndex is published so client knows commit channel is current
 	if len(ents) > 0 {
 		rc.lastIndex = ents[len(ents)-1].Index
-	} else {
-		rc.commitC <- nil
```
Comment (gyuho) [inline contrib/raftexample/raft.go]: why do we delete this? We need this to signal and trigger snapshot

Comment (upccup) [inline contrib/raftexample/raft.go]: rc.commitC <- nil  is trigger to loadsnap, at this  is use to trigger loadsnap  at the first time when we start the kvstore. but now we will do loadsnap without the init signal, so i think this is not useful.

Comment (gyuho) [inline contrib/raftexample/raft.go]: Oh right I misread where the line was. LGTM. Defer to @heyitsanthony

**Comment by gyuho** [comment]
@upccup Also the commit title should be something like `raftexample: fix blocking on restart`.

Thanks.

**Comment by upccup** [comment]
@gyuho  updated :)

File: contrib/raftexample/kvstore_test.go (hunk 1 above)
Comment (xiang90) [inline contrib/raftexample/kvstore_test.go]: we should come up with a similar test to ensure snapshot works.

Comment (upccup) [inline contrib/raftexample/kvstore_test.go]: @xiang90 I agree. But now i have noe good idea about make a mock snaoshot without start raft node.

**Comment by xiang90** [comment]
@gyuho Can you take another look? thanks.

**Comment by garvitjuniwal** [comment]
@gyuho @xiang90  Was this ever fixed? None of PRs referenced here seem to have merged.

**Pull request closed without merging**